services/geocoder/lib/hires_geocoder.rb

```
@@ -83,25 +82,29 @@ def config
 
     def http_client
       @http_client ||= Carto::Http::Client.get('hires_geocoder',
-        log_requests: true,
-        connecttimeout: HTTP_CONNECTION_TIMEOUT,
-        timeout: HTTP_REQUEST_TIMEOUT
-        )
+                                               log_requests: true,
+                                               connecttimeout: HTTP_CONNECTION_TIMEOUT,
+                                               timeout: HTTP_REQUEST_TIMEOUT)
     end
 
     def input_rows
-      stdout, stderr, status  = Open3.capture3('wc', '-l', input_file)
+      stdout, _stderr, _status = Open3.capture3('wc', '-l', input_file)
       stdout.to_i
-    rescue => e
+    rescue
       0
     end
 
     def process_row(input_row, output_csv_file)
-      @processed_rows = @processed_rows + 1
+      @processed_rows += 1
       latitude, longitude = geocode_text(input_row["searchtext"])
-      if !(latitude.nil? || latitude == "" || longitude.nil? && longitude == "")
+      if !(latitude.nil? || latitude == "") && !(longitude.nil? || longitude == "")
+        @successful_processed_rows += 1
         output_csv_file.add_row [input_row["searchtext"], 1, 1, latitude, longitude]
+      else
+        @empty_processed_rows += 1
       end
+    rescue
+      @failed_processed_rows += 1
```

I'd re-throw the error here or at least log it. Otherwise we might be swallowing errors that we did not intend to.